src/core/plugin/luapi_application.h

```diff
@@ -811,6 +812,105 @@ static int applib_addStrokes(lua_State* L) {
     return 0;
 }
 
+/**
+ * Puts a Lua Table of the coordinates of the points of all strokes from the selected layer/selection tool onto the
+ * stack
+ * TODO document the returned table and the parameters
+ */
+static int applib_getStrokes(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    std::string type = luaL_checkstring(L, 1);
+    std::vector<Element*> elements = {};
+    Control* control = plugin->getControl();
+
+    if (type == "Layer") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            control->clearSelection();  // otherwise strokes in the selection won't be recognized
+        }
+        elements = control->getCurrentPage()->getSelectedLayer()->getElements();
+    } else if (type == "Selection") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            elements = sel->getElements();
+        } else {
+            g_warning("There is no selection! ");
+            return 0;
+        }
+    } else {
+        g_warning("Unknown argument: %s", type.c_str());
+        return 0;
+    }
+
+    lua_newtable(L);  // create table of the elements
+    int currStrokeNo = 0;
+    int currPointNo = 0;
+
+    for (Element* e: elements) {
+        if (e->getType() == ELEMENT_STROKE) {
+            auto* s = dynamic_cast<Stroke*>(e);
+            lua_pushnumber(L, ++currStrokeNo);  // index for later (settable) TODO move down to the settable call?
+            lua_newtable(L);                    // create stroke table
+
+            lua_newtable(L);       // create table of x-coordinates
+            for (auto p: s->getPointVector()) {
+                lua_pushnumber(L, ++currPointNo);
+                lua_pushnumber(L, p.x);
+                lua_settable(L, -3);  // pops key and value from stack
+            }
+            lua_setfield(L, -2, "x");  // add x-coordinates to stroke
+            currPointNo = 0;
+
+            lua_newtable(L);       // create table for y-coordinates
+            for (auto p: s->getPointVector()) {
+                lua_pushnumber(L, ++currPointNo);
+                lua_pushnumber(L, p.y);
+                lua_settable(L, -3);
+            }
+            lua_setfield(L, -2, "y");  // add y-coordinates to stroke
+            currPointNo = 0;
+
+            if (s->hasPressure()) {
+                lua_newtable(L);  // create table for pressures
+                for (auto p: s->getPointVector()) {
+                    lua_pushnumber(L, ++currPointNo);
+                    lua_pushnumber(L, p.z);
+                    lua_settable(L, -3);
+                }
+                lua_setfield(L, -2, "pressure");  // add pressures to stroke
+                currPointNo = 0;
+            }
+
+            // TODO split up like addStrokesHelper?
```

Comment:
The idea of splitting `addStrokesHelper` off was to avoid code duplication (it is used in `addStrokes` and `addSplines`). There is currently no need to do so here.

Reply:
Yes, didn't wanted to think about this at the moment I wrote/adapted the function. Just thought maybe if we want to do something like `getTexts` there might be some code which can be shared, but at first glimpse only the color attribute is kinda shared (inherited from `Element`).
So yes I agree there is no need to extract this code (at the moment, and probably won't be without other major changes).